### PR TITLE
Add modkit pileup

### LINKED ISSUE #3950: new module: modkit/pileup
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by Michal-Babins**
## PR checklist

Closes #3950  

- [x ] This comment contains a description of changes (with reason).
- [ x] If you've fixed a bug or added code that should be tested, add tests!
- [ x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ x] If necessary, include test data in your PR.
- [ x] Remove all TODO statements.
- [ x] Emit the `versions.yml` file.
- [ x] Follow the naming conventions.
- [ x] Follow the parameters requirements.
- [ x] Follow the input/output options guidelines.
- [x ] Add a resource `label`
- [x ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/modkit/pileup/main.nf [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    input:
+    tuple val(meta), path(bam)
+    path bai
+    path out_bed
+
+    output:
+    tuple val(meta), path(out_bed), emit: bed
+    tuple val(meta), path("*_pileup.log"), emit: log
+    path "versions.yml"           , emit: versions
```
[the remark this answers is not in the document]
Comment (sateeshperi) [reply modules/nf-core/modkit/pileup/main.nf:18]: https://nf-co.re/docs/contributing/modules#what-is-the-harshil-alignment

File: tests/modules/nf-core/modkit/pileup/main.nf [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
```
[the remark this answers is not in the document]
Comment (SPPearce) [reply tests/modules/nf-core/modkit/pileup/main.nf:3]: No, we don't need pytest when nf-test exists

**Comment by SPPearce** [comment]
@Michal-Babins , do you need a hand to finish this module off?

**Comment by fellen31** [comment]
Sure, I'll fix it during the summer! @SPPearce

**Comment by SPPearce** [comment]
@fellen31 , I think it should only be a case of excluding the log file from the snapshots to be honest, shouldn't take very long to do.

**Comment by SPPearce** [comment]
Closing in favour of #5746

**Pull request closed without merging**